include/mbedtls/check_config.h

```diff
@@ -56,7 +56,7 @@
 
 /* Check that each MBEDTLS_ECP_DP_xxx symbol has its PSA_WANT_ECC_xxx counterpart
  * when PSA crypto is enabled. */
-#if defined(MBEDTLS_PSA_CRYPTO_CONFIG) || defined(MBEDTLS_PSA_CRYPTO_C)
+#if defined(MBEDTLS_PSA_CRYPTO_CONFIG) || defined(MBEDTLS_PSA_CRYPTO_CLIENT)
```

Comment:
Pre-existing: I don't think we should have this kind of checks here. We should just ensure that it's true by auto-enabling what we want in the relevant `config_adjust_` file.

Reply:
From `config_adjust_legacy_crypto.h` I can see this type of enablement:

```
/* Helpers to state that each key is supported either on the builtin or PSA side. */
#if defined(MBEDTLS_ECP_DP_SECP521R1_ENABLED) || defined(PSA_WANT_ECC_SECP_R1_521)
#define MBEDTLS_ECP_HAVE_SECP521R1
#endif
```

does this mean that the checks here in `check_config.h` can just be removed, because the `_HAVE` macro will be defined anyway when adjusting? If it is that simple, I think I can just remove the checks as part of this PR, but if it has wider implication, I think it would be better assessed with a dedicated PR.

Reply:
I'm removing those checks in https://github.com/Mbed-TLS/mbedtls/pull/8802